src/lib.rs

```diff
@@ -450,11 +450,8 @@ impl Cascade {
             return Err(CascadeError::Parse("missing filters"));
         }
 
-        if top_hash_alg.is_none() {
-            return Err(CascadeError::Parse("missing hash algorithm"));
-        }
-
-        let hash_algorithm = top_hash_alg.unwrap();
```

Comment:
Actually, looking at this again, we could do `let hash_algorithm = top_hash_alg.ok_or_else(|| CascadeError::Parse("missing hash algorithm"))?;` but it's not a big deal.